internal/datastore/proxy/indexcheck/indexcheck_test.go

```diff
@@ -66,3 +71,289 @@ func TestIndexCheckingFoundIndex(t *testing.T) {
 		require.NoError(t, err)
 	}
 }
+
+func TestWrapWithIndexCheckingDatastoreProxyIfApplicable(t *testing.T) {
+	t.Run("wraps SQL datastore", func(t *testing.T) {
+		ds := fakeDatastore{}
+		wrapped := WrapWithIndexCheckingDatastoreProxyIfApplicable(ds)
+		require.IsType(t, &indexcheckingProxy{}, wrapped)
+	})
+
+	t.Run("returns non-SQL datastore as-is", func(t *testing.T) {
+		ds := &nonSQLDatastore{}
+		wrapped := WrapWithIndexCheckingDatastoreProxyIfApplicable(ds)
+		require.Equal(t, ds, wrapped)
+	})
+}
+
+func TestIndexCheckingProxyMethods(t *testing.T) {
+	ds := fakeDatastore{}
+	proxy := NewIndexCheckingDatastoreProxy(ds).(*indexcheckingProxy)
```

Comment:
I think `NewIndexCheckingDatastoreProxy`  should be made private, it's not used outside of `indexcheck`, and perhaps it's dangerous for callers to use it